task/source-build/0.1/source-build.yaml

```diff
@@ -0,0 +1,472 @@
+                manifest_data = json.load(f)
+
+            for layer in manifest_data["layers"]:
+                digest = layer["digest"].split(":")[-1]
+                log.debug("untar layer %s", digest)
+
+                blob_member = ""
+                symlink_member = ""
+                with tarfile.open(f"{base_sources_extraction_dir}/{digest}", "r:gz") as tar:
+                    for member in tar:
+                        if member.isfile():
+                            blob_member = member.name
+                        elif member.issym():
+                            symlink_member = member.name
+
+                if symlink_member.startswith("./rpm_dir/"):
+                    dest_dir = sib_dirs.rpm_dir
+                    log.debug("Prepare SRPM %s", symlink_member)
+
+                    extract_blob_member(digest, blob_member, dest_dir,
+                                        rename_to=os.path.basename(symlink_member),
+                                        work_dir=base_sources_extraction_dir,
+                                        log=log)
+
+                elif symlink_member.startswith("./extra_src_dir/"):
+                    extra_src_archive = os.path.basename(symlink_member)
+                    log.debug("Prepare extra source %s", extra_src_archive)
+                    # one extra source archive per directory, no matter what the directory name is.
+                    dest_dir = create_dir(extra_src_dir, extra_src_archive)
+                    sib_dirs.extra_src_dirs.append(dest_dir)  # save this directory for bsi
+
+                    extract_blob_member(digest, blob_member, str(dest_dir),
+                                        rename_to=extra_src_archive,
+                                        work_dir=base_sources_extraction_dir,
+                                        log=log)
+
+                    # FIXME: perhaps the dependency archive (the Cachi2) might be handled differently
+                    run(["tar", "xvf", extra_src_archive], check=True, cwd=dest_dir)
+                    os.unlink(f"{dest_dir}/{extra_src_archive}")
+
+                else:
+                    log.warning("No known operation happened on layer %s", digest)
+
+            return len(os.listdir(sib_dirs.rpm_dir)) > 0 or len(sib_dirs.extra_src_dirs) > 0
+
+
+        def gather_prefetched_sources(work_dir: str, cachi2_dir: str, sib_dirs: SourceImageBuildDirectories) -> bool:
+            log = logging.getLogger("source-build.prefetched-sources")
+
+            # Guess if hermetic build is enabled
+            # NOTE: this guess does depend on how cachi2 runs inside prefetch-dependencies task.
+            cachi2_output_dir = f"{cachi2_dir}/output"
+            cachi2_env = f"{cachi2_dir}/cachi2.env"
+
+            has_cachi2_output = os.path.exists(cachi2_output_dir) and os.path.isdir(cachi2_output_dir)
+            has_cachi2_env = os.path.exists(cachi2_env) and os.path.getsize(cachi2_env) > 0
```

Comment:
cachito env file can be actually empty depending on pkg managers used and their needs

Reply:
Fixed.